zerver/views/registration.py

```diff
@@ -57,6 +57,13 @@ def check_prereg_key_and_redirect(request: HttpRequest, confirmation_key: str) -
     except ConfirmationKeyException as exception:
         return render_confirmation_key_error(request, exception)
 
+    if request.GET.get("support", None) and request.user.is_staff:
```

Comment:
This is not reqired.  Just pass `set_status=False` to get_object_from_key above in that case.